Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -92,6 +93,9 @@ enum MacroExpansionError: Error, CustomStringConvertible {
 
     case .preambleWithoutBody:
       return "preamble macro cannot be applied to a function with no body"
+
+    case .circularExpansion(let type, let syntax):
+      return "circular expansion detected: '\(syntax)' with macro implementation type '\(type)'"
```

Comment:
Note: I think the phrasing here could be improved. Maybe not in this PR.

Reply:
> FWIW the compiler’s error message is `recursive expansion of macro 'MacroName'`. I think we should copy that.

I think including both the macro name and type in the error message is more helpful for debugging?

Reply:
I think usually the mating between macro name and macro type is quite obvious, so I don’t think we need to include both. But I don’t have very strong opinions here.